token/program-2022/src/processor.rs

```diff
@@ -378,25 +381,31 @@ impl Processor {
                     account_info_iter.as_slice(),
                 )?
             }
-            (COption::Some(ref delegate), _) if cmp_pubkeys(authority_info.key, delegate) => {
+            (
+                PodCOption {
+                    option: PodCOption::<Pubkey>::SOME,
+                    value,
```

Comment:
nit: I think this was clearer in the original code:
```suggestion
                    value: delegate,
```